app/components/agenda/agenda-header/agenda-version-actions.js

```diff
@@ -176,6 +178,36 @@ export default class AgendaAgendaHeaderAgendaVersionActions extends Component {
     return approvedAgendaitems;
   }
 
+  regenerateDecisionReportNames = task(async () => {
+    const reports = await this.store.queryAll('report', {
+      'filter[:has-no:next-piece]': true,
+      'filter[:has:piece-parts]': true,
+      'filter[decision-activity][treatment][agendaitems][agenda][created-for][:id:]':
+        this.args.meeting.id,
+    });
+    if (reports?.length > 0) {
+      let { alterableReports } = await this.decisionReportGeneration.getAlterableReports(reports);
+      if (alterableReports.length === 0) {
+        this.toaster.error(
+          this.intl.t('reports-cannot-be-altered')
+        );
+        return;
+      }
+      await Promise.all(alterableReports.map(async (report) => {
+        const agendaitem = await this.store.queryOne('agendaitem', {
+          'filter[:has-no:next-version]': true,
+          'filter[treatment][decision-activity][report][:id:]': report.id,
+        });
+        const documentContainer = await report.documentContainer;
+        const pieces = await documentContainer.pieces;
+        report.name = await generateReportName(agendaitem, this.args.meeting, pieces.length);
+        await report.belongsTo('file').reload();
+        await report.save();
+      }));
```

Comment:
this part is a carbon copy of what happens in the `edit-meeting-modal`.
So it works yes, but I will probably move this to a service call then.
something like 
`this.decisionReportGeneration.regenerateDecisionReportNamesForMeeting.perform(this.args.meeting.id);` (the name might need work)
followed by actual regeneration of the reports similar to `edit-meeting-modal` (need to see)